src/main.rs

```diff
@@ -663,89 +1029,343 @@ fn get_setup(args: &[String]) -> Setup {
-            .map(|method| {
-                NormalisationMethod::from_str(method).expect("Invalid normalisation method")
-            })
-            .unwrap_or_default();
-        let normalisation_type = matches
-            .opt_str(NORMALISATION_GAIN_TYPE)
-            .as_deref()
-            .map(|gain_type| {
-                NormalisationType::from_str(gain_type).expect("Invalid normalisation type")
-            })
-            .unwrap_or_default();
-        let normalisation_pregain = matches
-            .opt_str(NORMALISATION_PREGAIN)
-            .map(|pregain| pregain.parse::<f64>().expect("Invalid pregain float value"))
-            .unwrap_or(PlayerConfig::default().normalisation_pregain);
-        let normalisation_threshold = matches
-            .opt_str(NORMALISATION_THRESHOLD)
-            .map(|threshold| {
-                db_to_ratio(
-                    threshold
-                        .parse::<f64>()
-                        .expect("Invalid threshold float value"),
-                )
-            })
-            .unwrap_or(PlayerConfig::default().normalisation_threshold);
-        let normalisation_attack = matches
-            .opt_str(NORMALISATION_ATTACK)
-            .map(|attack| {
-                Duration::from_millis(attack.parse::<u64>().expect("Invalid attack value"))
-            })
-            .unwrap_or(PlayerConfig::default().normalisation_attack);
-        let normalisation_release = matches
-            .opt_str(NORMALISATION_RELEASE)
-            .map(|release| {
-                Duration::from_millis(release.parse::<u64>().expect("Invalid release value"))
-            })
-            .unwrap_or(PlayerConfig::default().normalisation_release);
-        let normalisation_knee = matches
-            .opt_str(NORMALISATION_KNEE)
-            .map(|knee| knee.parse::<f64>().expect("Invalid knee float value"))
-            .unwrap_or(PlayerConfig::default().normalisation_knee);
+
+        let normalisation_method;
+        let normalisation_type;
+        let normalisation_pregain;
+        let normalisation_threshold;
+        let normalisation_attack;
+        let normalisation_release;
+        let normalisation_knee;
+
+        if !normalisation {
+            for a in &[
+                NORMALISATION_METHOD,
+                NORMALISATION_GAIN_TYPE,
+                NORMALISATION_PREGAIN,
+                NORMALISATION_THRESHOLD,
+                NORMALISATION_ATTACK,
+                NORMALISATION_RELEASE,
+                NORMALISATION_KNEE,
+            ] {
+                if matches.opt_present(a) {
+                    warn!(
+                        "Without the `--{}` / `-{}` flag normalisation options have no effect.",
+                        ENABLE_VOLUME_NORMALISATION, ENABLE_VOLUME_NORMALISATION_SHORT,
+                    );
+                    break;
+                }
+            }
+
+            normalisation_method = PlayerConfig::default().normalisation_method;
```

Comment:
You could DRY this up by doing a `let default_config = PlayerConfig::default()` first.

Reply:
Can do.

Reply:
Done.